### ISSUE TITLE
Master Monitor over other Monitors

### ISSUE THREAD

**Issue by maxwai** [louislam/uptime-kuma#2724]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Be able to set a kind of "Master" Monitor. So when this "Master" Monitor (this could be for example the Server itself where the Services are running) goes down, the other Monitors that depend on this "Master" Monitor go down as well.

This would allow to avoid sending multiple Notifications when the "Master" Monitor is down, since currently one would get a Notification for every Service that goes down and the Server itself.

This can be quite useful for people that want to check the individual Services (so to check if a docker goes down for example) but also to quickly see that the Server itself has a Problem (has crashed or doesn't have an Internet connection anymore).

Here is an example how this could look like: (ignore the Services in Maintenance)

![image](https://user-images.githubusercontent.com/44324946/216930635-81073cb2-5b3a-4fd6-bf83-de5a25a3467b.png)

Here the all Services (called "Dienste" here) are living on the Server so if the first Monitor goes down, all Services should also be marked as down immediately and only one Notification should be send instead of 6

### ✔️ Solution

Add a way to link multiple Monitors to a single Monitor that acts like a "Master Switch"

**Comment by drewlsvern** [comment]
I am also looking for a feature similar to this. I would be willing to contribute towards this feature. I would love to gather some thoughts from others on this before starting though.

I have used Nagios before so my mind immediately kind of jumped to how I used it. You have a host and then you have a bunch of services that belong to that host.

I'm wondering if it might be easiest to automatically setup a master host to ping and then any sub monitors could be anything else.

In terms of the UI, my thought is to have an accordion and then any sub monitors would be visible by expanding the accordion.

**Comment by uptimejeff** [comment]
If internet is down for the uptimekuma host, I'd like it to pause all notifications until internet is restored.
Example: ping a known host which ALL checks depend on (ping an upstream router or 8.8.8.8) or two hosts (1.1.1.1/8.8.8.8), if unreachable, pause all checks/notification until service restored.

**Comment by CommanderStorm** [comment]
I think this is a duplicate of https://github.com/louislam/uptime-kuma/issues/2487 or https://github.com/louislam/uptime-kuma/pull/1236
If you agree, could you please close this Issue, as duplicates only create immortal zombies and are really hard to issue-manage?
If not, what makes this issue unique enough to require an additional issue? (Could this be integrated into the issue linked above?) ^^

**Comment by CommanderStorm** [comment]
@maxwai 
forgot the ping in the above message, sorry

**Comment by maxwai** [comment]
@CommanderStorm
I looked at the issued, and you are right, this is a duplicate of #1236 . Sorry, didn't see that one.
